### ISSUE TITLE
Which other data do you need in the plots?

### ISSUE THREAD

**Issue by stared** [stared/livelossplot#18]
I am thinking what is the best amount of data displayed?
Maybe it is good to add other features - like the numerical value of current metrics, or maximal values.

Though, I don't want to make it too cluttered. So, I am interested in your feedback what is missing (or what you would consider more a distraction).

**Comment by minghao-wu** [comment]
Some of them might be beyond the scope of this package.

1. add some light grid lines in the plot.
2. allow us to add customized horizontal and vertical bars in the plot. It is very common for users to study if their model outperforms the baseline model and how fast their model converges.
3. A countdown timer.
4. live GPU utilization.
5. maxima and minima

BTW, I attempted to plot the loss curve at batch-level granularity. I found it is easy to do so when training, but kinda difficult to do it in evaluation stage. Can you provide an example? I'm using PyTorch.

**Comment by stared** [comment]
> add some light grid lines in the plot.

Use can set matplotlib style https://matplotlib.org/tutorials/introductory/customizing.html. For more/different styles, use https://seaborn.pydata.org/tutorial/aesthetics.html#seaborn-figure-styles.

> allow us to add customized horizontal and vertical bars in the plot. It is very common for users to study if their model outperforms the baseline model and how fast their model converges.

Interesting idea. I will consider that.

> A countdown timer.

Not sure. It only works if it knows the total number of batches/epochs. 

> live GPU utilization

You can add this data with in the `update`. Though, you would need to use some library getting this data (I am not sure how to do it).

> maxima and minima

Was thinking about it.

> BTW, I attempted to plot the loss curve at batch-level granularity.

I don't recommend that. Plot-drawing is a costly step, and this thing is likely to severely impact performance.  For evaluation - I am not sure if you want that. These plots work well for the same x axis for both train and eval.

**Comment by duhaime** [comment]
Hi @stared  thanks for this great project!

One nice feature would be if one could pass the PlotLossesKeras constructor a parameter `skip` that would skip the first `n` frames. If a cost function drops multiple orders of magnitude, even the log scaled y axis can make it harder to see the learning rate. Skipping the first n frames would allow one to get around this problem...

**Comment by stared** [comment]
@duhaime A good idea, struggled with the same thing myself. 

My idea was to specify some "rescaling" options, e.g. ignoring the first `n` or taking the last `n`. If you want to implement that, I would be more than happy to accept a Pull Request.

**Comment by stared** [comment]
@duhaime Added `skip_first` option (e.g. `plot_losses = PlotLossesCallback(skip_first=2)`) with 6c88783c52d341f928ad83f66c8f3622a4da023b. Should work with versions `0.4.0+`.

**Comment by dchen327** [comment]
> @duhaime Added `skip_first` option (e.g. `plot_losses = PlotLossesCallback(skip_first=2)`) with [6c88783](https://github.com/stared/livelossplot/commit/6c88783c52d341f928ad83f66c8f3622a4da023b). Should work with versions `0.4.0+`.

When I try using `skip_first`, I get `__init__() got an unexpected keyword argument 'skip_first'`.  I've tried all of these:
```
from livelossplot import PlotLossesKerasTF
from livelossplot.inputs.tf_keras import PlotLossesCallback
from livelossplot import PlotLosses
```

However, initializing gives the error above.

**Comment by Naich** [comment]
The skip_first feature seemed removed after 0.5 release.

**Comment by dchen327** [comment]
> The skip_first feature seemed removed after 0.5 release.

Is there a new feature added that does the same thing?
